### PR TITLE
Add jscs-jsdoc plugin

### LINKED ISSUE #994: Add jscs-jsdoc and cover the core with comments
Ref https://github.com/jscs-dev/node-jscs/pull/908#discussion_r24061948

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8271452-add-jscs-jsdoc-and-cover-the-core-with-comments?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by zxqfox**
Fixes #994

### CODE REVIEW

File: lib/checker.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ utils.inherits(Checker, StringChecker);
 /**
  * Loads configuration from JS Object. Activates and configures required rules.
  *
- * @param {Object} config
+ * @param {Object} [config]
```
Comment (zxqfox) [inline lib/checker.js]: Not sure how to resolve this ;-(

File: lib/config/generator.js [hunk 2]
```diff
@@ -97,7 +97,7 @@ Generator.prototype.generate = function(path) {
 
 /**
  * @private
- * @param  {Object[][]} resultsPerPreset - List of error objects for each preset's run of checkPath
+ * @param  {Array.<Object[]>} resultsPerPreset - List of error objects for each preset's run of checkPath
```
Comment (zxqfox) [inline lib/config/generator.js]: This is a bug in type parser: Kuniwak/jsdoctypeparser#14

File: lib/string-checker.js [hunk 3]
```diff
@@ -8,10 +8,10 @@ var MAX_FIX_ATTEMPTS = 5;
 
 /**
  * Starts Code Style checking process.
+ * Params are deprecated, should be removed in 2.0
  *
  * @name StringChecker
  *
- * Following params are deprecated, should be removed in 2.0
```
Comment (zxqfox) [inline lib/string-checker.js]: This line going to `@name` tag description. It's wrong.

File: lib/checker.js (hunk 1 above)
Comment (mrjoelkemp) [inline lib/checker.js]: If config is the only datum passed, I'd vote for removing the .apply and calling configure with config.

File: package.json [hunk 4]
```diff
@@ -87,6 +87,7 @@
     "browserify": "^9.0.3",
     "coveralls": "~2.11.2",
     "has-ansi": "~1.0.1",
+    "jscs-jsdoc": "^1.0.0",
```
Comment (mrjoelkemp) [inline package.json]: No ~?

**Comment by mrjoelkemp** [comment]
Lgtm. Nice job! Excited to see this land.

**Comment by mdevils** [comment]
Great job!

File: lib/checker.js (hunk 1 above)
Comment (zxqfox) [inline lib/checker.js]: @mrjoelkemp Nice tip. Fixing!

**Pull request closed without merging**